app/src/main/java/com/example/bigowlapp/activity/ViewAnotherUserActivity.java

```diff
@@ -1,112 +1,133 @@
         supRequestBtn = findViewById(R.id.SupRequest);
-        supRequestRef = FirebaseDatabase.getInstance().getReference().child("SupRequests");
+        currentUserID = authRepository.getCurrentUser().getUid();
+        noteTv = findViewById(R.id.note);
+        noteTv = findViewById(R.id.note);
+        noteTv2 = findViewById(R.id.note2);
+        String contactDetails = getIntent().getStringExtra("contactDetails");
+        noteText = "Contact: " + contactDetails + " is already registered to the application.";
+        noteTv.setText(noteText);
 
-        db.collection("users")
-                .whereEqualTo("phoneNumber", otherUserNumber)
-                .get()
-                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+        if (currentUserID.equals(otherUserID)) {
+            noteTv2.setText(noSelfRequest);
+            supRequestBtn.setClickable(false);
+            supRequestBtn.setText(canNotSend);
+        } else {
+            supRequestBtn.setClickable(true);
+            try {
+                observeRequests();
+                supRequestBtn.setOnClickListener(new View.OnClickListener() {
                     @Override
-                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
-                        if (task.isSuccessful()) {
-                            task.getResult();
-                            for (QueryDocumentSnapshot qds : task.getResult()) {
-                                otherUser = qds.toObject(User.class);
-                            }
-
-                        }
+                    public void onClick(View v) {
+                        DoRequest();
                     }
                 });
-
-        otherUserID = otherUser.getUId();
-        mUser = mAuth.getCurrentUser();
-        supRequestBtn.setOnClickListener(v -> DoRequest(otherUserID));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
     }
 
-    private void DoRequest(String otherUserID) {
-        if (supRequestStatus.equals("none")) {
-            HashMap<String, Object> hashMap = new HashMap<>();
-            hashMap.put("status", "pending");
-            supRequestRef.child(mUser.getUid()).child(otherUserID)
-                    .updateChildren(hashMap).addOnCompleteListener(task -> {
-                if (task.isSuccessful()) {
-                    Toast.makeText(ViewAnotherUserActivity.this,
-                            "You have sent a request to supervise ",
-                            Toast.LENGTH_SHORT).show();
-                    supRequestStatus = "sent_pending";
-                    supRequestBtn.setText(supBtn3);
-                } else {
-                    Toast.makeText(ViewAnotherUserActivity.this,
-                            "" + Objects.requireNonNull(task.getException()).toString(),
-                            Toast.LENGTH_SHORT).show();
-                }
-            });
+    private void DoRequest() {
```

Comment:
Fix methoid naming convention as noted by SonarCloud.
Should be `doRequest()`

Reply:
it has been fixed